site-src/geps/gep-1709.md

```diff
@@ -37,8 +45,295 @@ which implementations can prove they satisfy and be recognized for.
+    test := tests.ConformanceTests[i]
+    test.Run(t, cSuite)
+}
+```
+
+> **NOTE**: In the `suite.Options` above it's still possible to add `SkipTests`
+> but when used in conjunction with `Profile` this will result in a report that
+> the profile is not valid for reporting. Implementations in this state may be
+> able to report themselves as "in progress", see the
+> [certification section](#certification) for details.
+
+Alternatively for an `Extended` conformance profile where not all of the
+features are implemented (as described in the [profiles](#profiles) section
+above):
+
+```console
+$ go test ./conformance/... -args \
+    -gateway-class=acme \
+    -conformance-profile=Layer7Extended \
+    -unsupported-features=HTTPResponseHeaderModification,HTTPRouteMethodMatching,HTTPRouteQueryParamMatching,
+```
+
+Or the equivalent configuration using the Golang library:
+
+```go
+cSuite, err := suite.New(suite.Options{
+    GatewayClassName: "acme",
+    Profile: Layer7ExtendedProfile,
+    UnsupportedFeatures: sets.New(
+        suite.SupportHTTPResponseHeaderModification,
+        suite.SupportHTTPRouteMethodMatching,
+        suite.SupportHTTPRouteQueryParamMatching,
+    ),
+    // other options
+})
+require.NoError(t, err, "misconfigured conformance test suite")
+cSuite.Setup(t)
+
+for i := 0; i < len(tests.ConformanceTests); i++ {
+    test := tests.ConformanceTests[i]
+    test.Run(t, cSuite)
+}
+```
+
+> **NOTE**: `UnsupportedFeatures` must match features that are extended. This
+> option is invalid when used on a `Core` type profile.
+
+> **NOTE**: In the future we may consider expanding the options to configure
+> and run the conformance test suite using a container image and/or via a
+> configuration file, however it was decided to hold off from doing any of
+> those in this iteration to avoid it getting to big (see the
+> [alternatives](#alternatives-considered) section for more thoughts).
+
+Once an implementation has integrated with the conformance test suite, they can
+move on to [certification](#certification) to report the results.
+
+[go-test]:https://go.dev/doc/tutorial/add-a-test
+[go]:https://go.dev
+[lib]:https://pkg.go.dev/sigs.k8s.io/gateway-api@v0.6.1/conformance/utils/suite
+
+### Certification
+
+Implementations will be able to report their conformance testing results using
+our [reporting process](#reporting-process). Implementations will be able to
+visibly demonstrate their conformance results on their downstream projects and
+repositories using our [certification process](#certification-process).
+
+#### Reporting Process
+
+When conformance tests complete a log message will be emitted:
+
+```console
+Success! Conformance profile "Layer7Extended" passed for version v0.7.0
+The following extended features were not supported:
+- HTTPResponseHeaderModification
+- HTTPRouteQueryParamMatching
+```
+
+For the above output upstream Gateway API can be provided the following report:
```

Comment:
One thing I have not found in the doc is who is responsible for generating `ConformanceReport` from the log?

Reply:
gw-api go library? 
a shell script?

I am thinking it will probably be the first one but probably worth calling it out

Reply:
I was actually thinking for the first iteration of this, these could just be manually edited as they have straightforward contents and need to go through a review process anyway. Let's make that more explicit:

```suggestion
For the above output upstream Gateway API can manually generate or update a report
in the following format:
```